_server/src/BusinessLogic/UsersModule.cpp

```diff
@@ -123,3 +143,92 @@ ResponseData UsersModule::registerUser(const Dataset &request)
 
     return response;
 }
+
+ResponseData UsersModule::getRoles()
+{
+    SPDLOG_TRACE("UsersModule::getRole");
+    ResponseData response;
+    auto rolesVec = m_rolesRepository->getAll();
+
+    if (rolesVec.empty())
+    {
+        response.dataset[Keys::Role::ID] = {""};
+        response.dataset[Keys::Role::NAME] = {""};
+        SPDLOG_WARN("UsersModule::getRoles | array of roles is empty");
+    }
+    else
+    {
+        for (auto i = rolesVec.cbegin(); i != rolesVec.cend(); i++)
+        {
+            response.dataset[Keys::Role::ID].emplace_back(i->id);
+            response.dataset[Keys::Role::NAME].emplace_back(i->name);
+        }
+    }
+
+    return response;
+}
+
+ResponseData UsersModule::addNewRole(const Dataset &request)
+{
+    SPDLOG_TRACE("UsersModule::addRoles");
+    ResponseData response;
+    std::string roleName{};
+    try
+    {
+        roleName = request.at(Keys::Role::NAME).front();
+    }
+    catch (const std::out_of_range &e)
+    {
+        SPDLOG_WARN("UsersModule::addNewRole username or password does not exit");
+        throw ServerException(_M, "Server got no username or password in Dataset");
+    }
+    auto usersVec = m_rolesRepository->getByField(Keys::Role::NAME, roleName);
+    if (usersVec.empty())
+    {
+        m_rolesRepository->add(Role("", roleName));
```

Comment:
wtf is this?
`add(Role`
COMPILE AND RUN TESTS before you are asking for review

Reply:
done